localbot/AnonModule.cs

```diff
@@ -11,61 +11,102 @@ namespace localbot
+        //
+        // - command to set the history length and cooldown time for 
+        // newID and anonUser history respectivly
+
+        // Max ID number to be generated or chosen with >newID
         public const int maxID = 1000;
+        // Length of the history of IDs that each AnonUser records
         public const int historyLength = 5;
 
         private class AnonUser
         {
-            public ulong user { get; set; }
-            private Stack<int> ids;
-            public DateTime lastNewID { get; set; }
+            public ulong user;
+            private List<int> ids;
+            public DateTime lastNewID;
+            public bool blacklisted;
 
+            public bool timeout;
+            public DateTime timeoutEnd;
+
+            // Takes an unsigned long user and returns a AnonUser with 
+            // user set to the user
             public AnonUser(ulong user)
             {
                 this.user = user;
-                ids = new Stack<int>();
+                ids = new List<int>();
             }
 
+            // Returns the current ID number that the AnonUser has set
             public int getID()
             {
-                return ids.Peek();
+                return ids[0];
             }
 
-            public bool AliasAs(int id)
+            // Returns true if this AnonUser is blacklisted
+            public bool IsBlacklisted()
             {
-                bool foundID = false;
-                Stack<int> temp = new Stack<int>();
-                while(ids.Count != 0)
+                return blacklisted;
+            }
+
+            // Returns true if this user is timed out
+            public bool IsTimedout()
+            {
+                if (timeout == false)
                 {
-                    temp.Push(ids.Pop());
-                }
-                while (temp.Count != 0)
+                    return false;
+                } else
                 {
-                    int cur = temp.Pop();
-                    if (cur == id)
+                    if(timeoutEnd - DateTime.Now > new TimeSpan(0, 0, 0))
                     {
-                        foundID = true;
+                        return true;
+                    } else
+                    {
+                        return false;
                     }
-                    ids.Push(cur);
-                }                
-                return foundID;
+                }
+            }
+
+            // Returns true if this AnonUser has used int id as an
+            // id in historyLength ids
+            public bool AliasAs(int id)
+            {
+                return ids.Contains(id);
             }
 
+            // Changes the active ID for this AnonUser and pushes the
+            // old ids to the history
             public void NewAlias(int alias)
             {
-                ids.Push(alias);
+                if (ids.Count > historyLength)
```

Comment:
```suggestion
                if (ids.Count >= historyLength)
```

Right now, if ids has `historyLength` elements, then it will just add the element, making for `historyLength+1` elements.